src/FakeItEasy/Core/IFakeObject.cs

```diff
@@ -0,0 +1,28 @@
+﻿namespace FakeItEasy.Core
+{
+    /// <summary>
+    /// This interface is used internally by FakeItEasy and is exposed only due to technical constraints. It is not
+    /// intended to be used directly from your code.
+    /// </summary>

```

Comment:
Yes, it would probably make sense. `ITaggable` has always felt kind of wrong to me. It looks like it's designed for a very generic usage (the name `Tag` is vague, and it's typed as `object`), but we use it for something very specific.

Of course, if we could make both interfaces internal, it would be even better.